contracts/DCAStrategies/DCAStrategies/DCAStrategiesPositionsHandler.sol

```diff
@@ -226,15 +250,98 @@ abstract contract DCAStrategiesPositionsHandler is IDCAStrategiesPositionsHandle
   }
 
   function _calculateOptimalAmount(
-    uint256 _index,
-    uint256 _arrayLength,
+    bool _isLastOne,
     uint256 _amount,
     uint256 _share,
     uint256 _total,
     uint256 _amountSpent
   ) internal pure returns (uint256 _optimal) {
     if (_amount == 0) return 0;
-    return _index < _arrayLength - 1 ? (_amount * _share) / _total : _amount - _amountSpent;
+    // if isn't the last one, assign the share of amount. If it's the last one, assign the leftover
+    return !_isLastOne ? (_amount * _share) / _total : _amount - _amountSpent;
+  }
+
+  function _syncFirstBlock(
+    Position memory _position,
+    uint256 _totalAmount,
+    IDCAStrategies.ShareOfToken[] memory _newTokenShares,
+    uint32 _newAmountSwaps,
+    address _recipientSwapped
+  ) internal {
+    uint8 _currentPositionsIndex; // old positions index
+    uint8 _newPositionsIndex; // new positions index
+    uint256 _totalRemaining; // cash. The amount of money ready to use. (Used to know if need to send whats left or request whats missing)
+    uint256 _amountSpent;
+    Task[] memory _tasks = new Task[](_newTokenShares.length); // an array containing required tasks
```

Comment:
The length here should be the sum of the amount of existing positions, and the amount of new token shares

Reply:
I remember in the last meeting we discussed that the task array will be the same length as the new shares array

Reply:
Oh, you are right, because we won't be storing "terminate" actions. My bad